src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,89 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75F;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        size = 0;
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        if (addToTable(key, value, table)) {
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Node<K, V> nodeByKey = table[hash(key, table.length)];
+        while (nodeByKey != null) {
+            if ((key != null && key.equals(nodeByKey.key)) || key == nodeByKey.key) {
+                return nodeByKey.value;
+            }
+            nodeByKey = nodeByKey.nextNode;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean addToTable(K key, V value, Node<K, V>[] table) {
+        int hash = hash(key, table.length);
+        if (table[hash] == null) {
+            table[hash] = new Node<>(key, value, null);
+            return true;
+        } else {
+            Node<K, V> lastNode = table[hash];
+            while (lastNode != null) {
+                if ((lastNode.key != null && lastNode.key.equals(key))
+                        || lastNode.key == key) {
+                    lastNode.value = value;
+                    break;
+                } else if (lastNode.nextNode == null) {
```

Comment:
do we need this check?

Reply:
replaced it with if, and removed all from this method to put method.